flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -1314,15 +1323,15 @@ Fortran::lower::pft::FunctionLikeUnit::FunctionLikeUnit(
       endStmt{getFunctionStmt<parser::EndMpSubprogramStmt>(func)} {
   auto symbol = getSymbol(*beginStmt);
   entryPointList[0].first = symbol;
-  processSymbolTable(*symbol->scope(), varList);
+  processSymbolTable(*symbol->scope(), varList, /*reentrant=*/false);
```

Comment:
Why are separate module program considered to always be non-reentrant ?

Reply:
This may be a bug. I marked modules as not reentrant, and this got caught up with that.